src/builder.js

```diff
@@ -569,6 +574,49 @@ export default function builder() {
       };
     });
 
+    instrument(window, 'onunhandledrejection', function(original) {
+      // See https://developer.mozilla.org/en-US/docs/Web/API/Window/unhandledrejection_event
+      function onunhandledrejection(promiseRejectionEvent) {
+        debug('window.onunhandledrejection callback invoked', arguments);
+
+        // Skip if the error is already being sent.
+        if (currentErr) { return; }
+
+        if (!onUnhandledRejectionEnabled()) { return; }
+
+        let { reason } = promiseRejectionEvent;
+
+        if (reason instanceof Error) {
+          // simulate v8 stack
+          let fileName = reason.fileName || 'unknown';
+          let lineNumber = reason.lineNumber || 0;
+          let stackFallback = `${reason.message}\n    at ? (${fileName}:${lineNumber})`;
+          let stack = stackTrace(reason) || stackFallback;
+
+          notify({
+            name: reason.name,
+            message: `UnhandledPromiseRejectionWarning: ${reason}`,
+            stack
+          });
+
+          return;
+        }
+
+        let message = typeof reason === 'string' ? reason : JSON.stringify(reason)
+        notify({
```

Comment:
Also had to tweak the `notify` call here, to not rely on the `generated` argument (which is reserved for some special cases).